api/product.rb

```diff
@@ -39,17 +40,27 @@ def validate
 
     # Represents a version of the API for this product
     class Version < Api::Object
+      include Comparable
       attr_reader :base_url
       attr_reader :default
       attr_reader :name
 
+      ORDER = %w[v1 beta alpha].freeze
+
       def validate
         super
         @default ||= false
 
         check_property :base_url, String
         check_property :name, String
         check_property :default, :boolean
+
+        raise "API Version must be one of #{ORDER}" \
+          unless ORDER.include?(@name)
+      end
+
+      def <=>(other)
+        ORDER.index(name) <=> ORDER.index(other.name) if other.is_a?(Version)
```

Comment:
Can we get a exception raised here if other is not a Version?

If not, this function will return nil, which could reek havoc.

```ruby
def <=>(other)
  raise "#{other} is not a version" unless other.is_a? Version
  ORDER.index(name) <=> ORDER.index(other.name)
```

Reply:
Done!